test/data/test_builtin_datasets.py

```diff
@@ -76,6 +76,26 @@ def test_raw_text_classification(self, info):
             self.assertEqual(torchtext.datasets.MD5[dataset_name], info['MD5'])
         del data_iter
 
+    @parameterized.expand(
+        load_params('raw_datasets.jsonl'),
+        name_func=_raw_text_custom_name_func)
+    def test_dataset_iterator_offsets(self, info):
```

Comment:
Do we need a new unit test or can we modify the existing one [here](https://github.com/pytorch/text/blob/master/test/data/test_builtin_datasets.py#L58)?

Reply:
I thought about this too, but I felt it would be it good to not cram many checks into one unit test. If you think it's better to just modify the existing one, let me know.

Reply:
As such after introducing new modifications in raw_datasets.jsonl w.r.t offsets, I do not see any difference between the two unit tests for the part where we validate hash. I think it's ok to keep this part in a separate unit test. If so, we probably want to remove line verification in the other function.

Reply:
The line verification in the other function is specifically for the first line though. Once we make the offsets non-zero for the rest of the datasets wouldn't we want to keep the first line verification as is?

Reply:
isn't it testing at offset=0 is same as what's being tested in other function?

Reply:
yeah. but when we make the offsets non-zero in the future the test for the first line won't happen anymore if we remove it from the existing test.

Reply:
I'd keep tests for all the datasets/splits with offset=0, and keep adding lines in raws_datasets.jsonl with non-zero offsets for datasets that support offset.